### DISCUSSION TITLE
Isn't there a simple fix for the elevation required issue?

### DISCUSSION THREAD

**Discussion by BurtHarris** [coreybutler/nvm-windows#862] [Ideas & Feature Requests] [answered]
I know there's been lots of discussion around this before. Still, it looks to me like the root cause of current problems with NVM needing elevated permissions is the current definition of the NVM_SYMLINK environment variable pointing into the shared ProgramFiles folder and not into a per-user folder.

I have modified that (and the corresponding PATH section), and I've been happy with the result for months.  Here's a PowerShell script that does it (only for the current shell session): 

```powershell
$nvmhome = $env:NVM_HOME  # C:\Users\burt_\AppData\Roaming\nvm
$old = $env:NVM_SYMLINK   # C:\Program Files\nodejs
$new = $nvmhome.Replace('nvm','nodejs') 
$env:NVM_SYMLINK = $new   # C:\Users\burt_\AppData\Roaming\nodejs
$env:Path = $env:Path.Replace($old, $new) # not required, just in case
```
Is there a reason the current installer wouldn't work with this simple change?

I don't understand the InnoSetup stuff, or I would submit a PR.

**Comment by coreybutler** [comment, marked as answer]
The installer already provides the option to specify whatever directory you want. I usually just install to `C:\nvm4w`.

That said, your script seems like a good option for those who have "directory selection remorse" and want to make a change.

For most users, not having `NVM_SYMLINK` pointed to a protected directory would be enough. However; in some environments, symlinks themselves require elevated admin privileges, regardless of whether the directory is protected or not. This is prominent in older versions of Windows. Newer versions of Windows have "Developer Mode", which will help bypass elevation for symlinks on non-protected directories.

**Reply by BurtHarris** [reply]
Thanks, Corey.  I forgot I've got developer mode turned on.   I'm getting old like that.

**Reply by BurtHarris** [reply]
P.S.  I'm installing it with Winget.   It might be good if the installer detected if the no-elevation option was available and gave the user the option to install that way.

**Reply by coreybutler** [reply]
Just a reminder, the winget installer isn't part of this project. It might be worth bringing this up with the folks maintaining that installer.

**Reply by BurtHarris** [reply]
Oh!  Interesting Corey!  It looks like my thanks are due to @awdr74100 and @ItzLevvie for integrating NVM-Windows and Winget.    Perhaps they can add some detail here.

For anyone interested in looking, here is the [manifest file](https://github.com/microsoft/winget-pkgs/blob/master/manifests/c/CoreyButler/NVMforWindows/1.1.9/CoreyButler.NVMforWindows.installer.yaml).  Technically the manifest isn't an installer, the current version references `https://github.com/coreybutler/nvm-windows/releases/download/1.1.9/nvm-setup.exe`, and it looks like the manifest is [generated by a tool](https://github.com/microsoft/winget-create#welcome-to-the-windows-package-manager-manifest-creator-repository) thus it is probably not a good place to alter the default environment variable setting.

**Reply by BurtHarris** [reply]
The other change that could help is to use an _NTFS junction_ rather than a _symbolic link_.   I think this was discussed in #79.   

The windows architects implemented junction points as a more secure-by-default alternative.  [Comparison here](https://stackoverflow.com/questions/9042542/what-is-the-difference-between-ntfs-junction-points-and-symbolic-links#:~:text=What%20is%20the%20difference%20between%20NTFS%20Junction%20Points,while%20SymLinks%20are%20allowed%20to%20also%20target%20files.).  There are specific scenarios one can invent that might need the flexibility of a full symbolic link. Still, I think use cases involving nonprivileged single-user installation would not need symlinks.

**Reply by BurtHarris** [reply]
💡 How about ensuring %NVM_SYMLINK% is a child of %NVM_HOME% ?

- NTFS Junctions would be local to NVM_HOME;  No cross-volume scenarios.   
- a large amount of the complications in the setup could go away
- no need for developer mode or admin.

**Reply by BurtHarris** [reply]
@coreybutler,  @awdr74100: there is a draft PR #865 with an update that eliminates the need for elevated execution for installation or use.  

This version currently does NOT support ALLUSERS installation or installation over an existing NodeJS install.   I haven't ripped that code out; I don't have the options enabled.   I started to test those scenarios but would like to discuss the desired behavior.
